### PR TITLE
DBZ-253 Avoiding exception when receiving DDL events with table maint…

### PR DESCRIPTION
**Pull request by gunnarmorling**
…enance statements

@rhauch this should fix [DBZ-253](https://issues.jboss.org/browse/DBZ-253) by letting the DDL parser gracefully handle table maintenance statements such as `ANALYZE`.

There's no specific handling of these (I don't think anything is needed?). the only thing I was wondering is whether those statements should show up in the schema change topic (currently they do). They don't actually change the schema, but then I reckon reporting events for them doesn't do harm. WDYT?

### CODE REVIEW

**Comment by rhauch** [comment]
@gunnarmorling They will be parsed as "unknown" statements (see [here](https://github.com/debezium/debezium/blob/master/debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlDdlParser.java#L157)) and [just consumed](https://github.com/debezium/debezium/blob/master/debezium-core/src/main/java/io/debezium/relational/ddl/DdlParser.java#L375). They don't need to go into history (filtering out would be extra work), but it's okay if they are.

**Comment by gunnarmorling** [comment]
@rhauch to be sure, we agree that this change is good, right? As you say, these statements are handled by `parseUnknownStatement()`, but before this change an exception was raised:

    io.debezium.text.ParsingException: Expecting token type 128 at line 1, column 1 but found 'ANALYZE':  ===>> ANALYZE TABLE `App`
    at io.debezium.text.TokenStream.consume(TokenStream.java:725)
    at io.debezium.relational.ddl.DdlParser.consumeStatement(DdlParser.java:567)
    at io.debezium.relational.ddl.DdlParser.parseUnknownStatement(DdlParser.java:375)

On the filtering, agreed it's ok to have them there. If someone wants to have them filtered out, we can do it later on. So I'd proceed and merge this PR.

File: debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDdlParserTest.java [hunk 1]
```diff
@@ -1031,6 +1030,20 @@ public void shouldParseCreateTableWithBitDefault() {
...
+    public void shouldParseTableMaintenanceStatements() {
+        String ddl = "create table `db1`.`table1` ( `id` int not null, `other` int );";
+        ddl += "analyze table `db1`.`table1`;";
+        ddl += "optimize table `db1`.`table1`;";
+        ddl += "repair table `db1`.`table1`;";
+
+        parser.parse(ddl, tables);
+        Testing.print(tables);
+        assertThat(tables.size()).isEqualTo(1);
+        assertThat(listener.total()).isEqualTo(1);
```
Comment (rhauch) [inline debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlDdlParserTest.java]: Probably would be nice to have these in an integration test.

**Comment by gunnarmorling** [comment]
@rhauch Added one more commit with an integration test. Merge it if you like?

**Comment by rhauch** [comment]
@gunnarmorling what should be merged first? This or PR #234?

**Comment by gunnarmorling** [comment]
> what should be merged first?

No preference, I'll do this one first and then rebase the other. I generally prefer rebasing over merge commits as it keeps the history nice and lean. Thanks for reviewing!

**Pull request merged**